src/common/helpers/snap-builds.js

```diff
@@ -25,43 +25,116 @@
 // };
 
 export const BuildStatusColours = {
+  BLUE: 'blue',
   GREEN: 'green',
   YELLOW: 'yellow',
   RED: 'red',
   GREY: 'grey'
 };
 
+const BuildAndPubState = {
+  'NEVER_BUILT':     createState('Never built', BuildStatusColours.GREY, false, 8),
+  'BUILDING_SOON':   createState('Building soon', BuildStatusColours.GREY, 'ellipsis', 7),
+  'IN_PROGRESS':     createState('In progress', BuildStatusColours.BLUE, 'spinner', 2),
+  'FAILED_TO_BUILD': createState('Failed to build', BuildStatusColours.RED, 'cross', 1),
+  'WONT_PUBLISH':    createState('Built, won\'t be published', BuildStatusColours.GREEN, 'tick_outlined', 6),
+  'PUBLISH_SOON':    createState('Built, publishing soon', BuildStatusColours.GREY, 'tick', 2),
+  'PUBLISH_NOW':     createState('Built, publishing now', BuildStatusColours.BLUE, 'tick', 3),
+  'PUBLISH_FAILED':  createState('Built, failed to publish', BuildStatusColours.RED, 'tick', 4),
+  'PUBLISHED':       createState('Built and published', BuildStatusColours.GREEN, 'tick_filled', 5)
+};
+
+function createState(message, colour, icon, priority) {
+  return {
+    message,
+    colour,
+    icon,
```

Comment:
Not sure if in the end we want to have purely UI stuff like colour and icon here, but on the other hand we already have colour here and it being on client side it can be treated as UI code, so maybe no need to worry about this now.

Especially that we don't have icons yet, so we can postpone working on that to the point when we will be adding the icons to UI.